.github/scripts/pattern_analyzer.py

```diff
@@ -0,0 +1,146 @@
+import re
+from enum import Enum
+from dataclasses import dataclass
+from typing import List
+
+
+class Confidence(Enum):
+    LOW = "low"
+    MEDIUM = "medium"
```

Comment:
I don't think we treat `LOW` and `MEDIUM` differently to each other. We probably only need `LOW` and `HIGH`.